src/widgets/WidgetMainQmc.vue

```diff
@@ -107,6 +116,95 @@ export default defineComponent({
+    getPrimaryProduct() {
+      this.primaryProduct.productType = this.products[0].productType
+      this.primaryProduct.productItems = this.products[0].productItems
+    },
+    getCardAccountFee() {
+      this.cardAccountFee = this.cards.find(
+        card => card.id === 'QMC1'
+      )?.annualFee
+    },
+    getCardStandardInterestRate() {
+      this.cardStandardInterestRate = this.cards.find(
+        card => card.id === 'QMC1'
+      )?.interestRate
+    },
+    getLongTermInterestFreeTitle() {
+      const interestFreePeriod = this.primaryProduct.productItems[0].contents.find(
+        item => item.key === 'interestFreePeriod'
+      )?.value
+      const indicativeMinMonthly = this.primaryProduct.productItems[0].contents.find(
+        item => item.key === 'indicativeMinMonthly'
+      )?.value
+      return `${interestFreePeriod} Month${
+        interestFreePeriod === 1 ? '' : 'ly'
+      } Payment${
+        interestFreePeriod === 1 ? '' : 's'
+      } of $${indicativeMinMonthly}*`
+    },
+    getPaymentHolidayTitle() {
+      const interestFreePeriod = this.primaryProduct.productItems[0].contents.find(
+        item => item.key === 'interestFreePeriod'
+      )?.value
+      return `${interestFreePeriod} Month${
+        interestFreePeriod === 1 ? '' : 's'
+      } No Payments & No Interest`
+    },
+    getLongTermInterestFreeSubtitle(
+      establishmentFee: number,
+      advancedFee: number
+    ) {
+      if (establishmentFee === 0 || advancedFee === 0)
+        return `*indicative. $${this.cardAccountFee} annual Account Fee, Ts&Cs apply.`
+      return `*indicative. $${establishmentFee} Establishment Fee or $${advancedFee} Advance Fee, $${this.cardAccountFee} annual Account Fee, Ts&Cs apply.`
+    },
+    getPaymentHolidaySubtitle(establishmentFee: number, advancedFee: number) {
+      if (establishmentFee === 0 || advancedFee === 0)
+        return `Standard Interest Rate ${this.cardStandardInterestRate}% applies at the end of the interest free period. $${this.cardAccountFee} annual Account Fee. Ts&Cs apply.`
+      return `Standard Interest Rate ${this.cardStandardInterestRate}% applies at the end of the interest free period. $${establishmentFee} Establishment Fee or $${advancedFee} Advance Fee applies. $${this.cardAccountFee} annual Account Fee. Ts&Cs apply.`
+    },
+    createTitle() {
+      switch (this.primaryProduct.productType) {
+        case ProductEnum.LongTermInterestFree:
+          this.title = this.getLongTermInterestFreeTitle()
+          break
+        case ProductEnum.PaymentHoliday:
+          this.title = this.getPaymentHolidayTitle()
+          break
+      }
+    },
+    createSubtitle() {
+      const establishmentFee = this.primaryProduct.productItems[0].contents.find(
+        item => item.key === 'establishmentFee'
+      )?.value as number
+      const advancedFee = this.primaryProduct.productItems[0].contents.find(
+        item => item.key === 'advancedFee'
+      )?.value as number
```

Comment:
You're doing a lot of `find` on `this.primaryProduct.productItems[0].contents` here and further up. Could it be a utility function, or handled in another way?

Reply:
new function created `getProductPropertyByValue`